jobs/workers.go

```diff
@@ -133,7 +133,8 @@ func (srv *JobServer) InitWorkers() error {
 // Synchronization should be implemented by the caller.
 func (workers *Workers) Start() {
 	mlog.Info("Starting workers")
-	if workers.DataRetention != nil && (*workers.ConfigService.Config().DataRetentionSettings.EnableMessageDeletion || *workers.ConfigService.Config().DataRetentionSettings.EnableFileDeletion) {
+
+	if workers.DataRetention != nil {
```

Comment:
That would require a DB call in order to know if there's anything configured in `RetentionPoliciesTeams` or `RetentionPoliciesChannels`. And any updates to those tables would require a hook to enable or disable the worker accordingly. So I think options include:
a) managing that state of whether the worker should run as described above, or
b) doing a single DB check each day for every instance (regardless of whether any retention policies are configured); that's how it's currently working, or
c) an on/off toggle in the UI

Reply:
@mkraft I think a) is the only viable option. Doing a single DB check once per day won't work if the sysadmin wants to test out a policy which they just created. And this feels too low-level to have a toggle for it in the UI (since the user would be exposed to implementation details)l

We could also just run an empty job, if the extra complexity isn't worth it.

Reply:
@maxerenberg I actually mispoke when I rote "each day" I meant to say every time the workers start. The current code. That's how it's working currently in the PR and I think that's the simplest implementation with barely any performance cost.